internal/servers/worker/handler.go

```diff
@@ -89,73 +91,86 @@ func (w *Worker) handleProxy() http.HandlerFunc {
 		// Later calls will cause this to noop if they return a different status
 		defer conn.Close(websocket.StatusNormalClosure, "done")
 
-		connCtx, connCancel := context.WithDeadline(r.Context(), expiration.AsTime())
+		connCtx, connCancel := context.WithDeadline(ctx, expiration.AsTime())
 		defer connCancel()
 
+		sessClient, err := w.ControllerSessionConn()
+		if err != nil {
+			event.WriteError(ctx, op, err)
+			_ = conn.Close(websocket.StatusInternalError, "unable to get controller session client")
```

Comment:
Any reason not to write the close error as an event if we hit one?

Reply:
Same question in the blocks below

Reply:
Not specifically this was more a linting exercise to explicitly deal with the unhandled error. Can update to write all these now that we have events

Reply:
Not sure if it's deemed too immaterial/inconsequential or not. Maybe a good question for @jimlambrt

Reply:
I think, we should probably WriteError(...) for the close error.  I noticed this when "porting" it to events and forgot to raise the question in my PR.